src/lib/viewers/controls/slider/SliderControl.tsx

```diff
@@ -1,42 +1,151 @@
     className?: string;
-    onChange?: (value: number) => void;
+    max?: number;
+    min?: number;
+    onMove?: (value: number, position: number, width: number) => void;
+    onUpdate?: (value: number) => void;
+    step?: number;
+    title: string;
     track?: string;
+    value: number;
 };
 
-export type Ref = HTMLInputElement;
+export type Ref = HTMLDivElement;
 
-function SliderControl({ className, onChange = noop, track, ...rest }: Props, ref: React.Ref<Ref>): JSX.Element {
-    const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>): void => {
-        onChange(parseFloat(target.value));
+export default function SliderControl({
+    className,
+    max = 100,
+    min = 0,
+    onMove = noop,
+    onUpdate = noop,
+    step = 1,
+    title,
+    track,
+    value,
+    ...rest
+}: Props): JSX.Element {
+    const [isScrubbing, setIsScrubbing] = React.useState(false);
+    const sliderElRef = React.useRef<Ref>(null);
+
+    const getPosition = React.useCallback((pageX: number) => {
+        const { current: sliderEl } = sliderElRef;
+
+        if (!sliderEl) return 0;
+
+        const { left: sliderLeft, width: sliderWidth } = sliderEl.getBoundingClientRect();
+        return Math.max(0, Math.min(pageX - sliderLeft, sliderWidth));
+    }, []);
+
+    const getPositionValue = React.useCallback(
+        (pageX: number) => {
+            const { current: sliderEl } = sliderElRef;
+
+            if (!sliderEl) return 0;
+
+            const { width: sliderWidth } = sliderEl.getBoundingClientRect();
+            const newValue = (getPosition(pageX) / sliderWidth) * max;
+            return Math.max(min, Math.min(newValue, max));
+        },
+        [getPosition, max, min],
+    );
+
+    const handleKeydown = (event: React.KeyboardEvent): void => {
+        const key = decodeKeydown(event);
+
+        if (key === 'ArrowLeft') {
+            event.stopPropagation(); // Prevents global key handling
+            onUpdate(Math.max(min, Math.min(value - step, max)));
+        }
+
+        if (key === 'ArrowRight') {
+            event.stopPropagation(); // Prevents global key handling
+            onUpdate(Math.max(min, Math.min(value + step, max)));
+        }
+    };
+
+    const handleMouseDown = ({ button, ctrlKey, metaKey, pageX }: React.MouseEvent<Ref>): void => {
+        if (button > 1 || ctrlKey || metaKey) return;
+
+        onUpdate(getPositionValue(pageX));
+        setIsScrubbing(true);
     };
 
+    const handleMouseMove = ({ pageX }: React.MouseEvent<Ref>): void => {
+        const { current: sliderEl } = sliderElRef;
+        const { width: sliderWidth } = sliderEl ? sliderEl.getBoundingClientRect() : { width: 0 };
+
+        onMove(getPositionValue(pageX), getPosition(pageX), sliderWidth);
+    };
+
+    const handleTouchStart = ({ touches }: React.TouchEvent<Ref>): void => {
+        onUpdate(getPositionValue(touches[0].pageX));
+        setIsScrubbing(true);
+    };
+
+    React.useEffect(() => {
```

Comment:
Are there any issues with this when doing a quick click and drag and release interaction with the slider?

Reply:
I haven't seen any issues in my testing, but will keep an eye out as we build more features with this component.